src/sqlfluff/core/linter/linted_file.py

```diff
@@ -0,0 +1,491 @@
+        return not any(self.get_violations(filter_ignore=True))
+
+    def fix_string(self) -> Tuple[Any, bool]:
+        """Obtain the changes to a path as a string.
+
+        We use the source mapping features of TemplatedFile
+        to generate a list of "patches" which cover the non
+        templated parts of the file and refer back to the locations
+        in the original file.
+
+        NB: This is MUCH FASTER than the original approach
+        using difflib in pre 0.4.0.
+
+        There is an important distinction here between Slices and
+        Segments. A Slice is a portion of a file which is determined
+        by the templater based on which portions of the source file
+        are templated or not, and therefore before Lexing and so is
+        completely dialect agnostic. A Segment is determined by the
+        Lexer from portions of strings after templating.
+        """
+        bencher = BenchIt()
+        bencher("fix_string: start")
+
+        linter_logger.debug("Original Tree: %r", self.templated_file.templated_str)
+        assert self.tree
+        linter_logger.debug("Fixed Tree: %r", self.tree.raw)
+
+        # The sliced file is contiguous in the TEMPLATED space.
+        # NB: It has gaps and repeats in the source space.
+        # It's also not the FIXED file either.
+        linter_logger.debug("### Templated File.")
+        for idx, file_slice in enumerate(self.templated_file.sliced_file):
+            t_str = self.templated_file.templated_str[file_slice.templated_slice]
+            s_str = self.templated_file.source_str[file_slice.source_slice]
+            if t_str == s_str:
+                linter_logger.debug(
+                    "    File slice: %s %r [invariant]", idx, file_slice
+                )
+            else:
+                linter_logger.debug("    File slice: %s %r", idx, file_slice)
+                linter_logger.debug("    \t\t\ttemplated: %r\tsource: %r", t_str, s_str)
+
+        original_source = self.templated_file.source_str
+
+        # Make sure no patches overlap and divide up the source file into slices.
+        # Any Template tags in the source file are off limits.
+        source_only_slices = self.templated_file.source_only_slices()
+
+        linter_logger.debug("Source-only slices: %s", source_only_slices)
+
+        # Iterate patches, filtering and translating as we go:
+        linter_logger.debug("### Beginning Patch Iteration.")
+        filtered_source_patches = []
+        dedupe_buffer = []
+        # We use enumerate so that we get an index for each patch. This is entirely
+        # so when debugging logs we can find a given patch again!
+        patch: Union[EnrichedFixPatch, FixPatch]
+        for idx, patch in enumerate(
+            self.tree.iter_patches(templated_str=self.templated_file.templated_str)
+        ):
+            linter_logger.debug("  %s Yielded patch: %s", idx, patch)
+
+            # This next bit is ALL FOR LOGGING AND DEBUGGING

```

Comment:
Nitpick: The need for this comment makes me wonder -- would it potentially be more readable to move all this code to a separate function?

Reply:
Yes 👍 . I just lifted and shifted this so far, but good idea.

Reply:
Heh, yeah, moving code around makes reviews tricky -- pre-existing code gets a fresh look. Don't stress too much about this. FWIW, some IDEs (such as PyCharm) have a "Refactor"/"Extract Method" menu choice that streamlines things like this. Very handy -- it's pretty smart about figuring out the parameters required by the function, etc.